internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -41,10 +51,17 @@ type HypershiftAgentServiceConfigReconciler struct {
 	// on the management cluster
 	NodeSelector map[string]string
 	Tolerations  []corev1.Toleration
+
+	// Namespace the operator is running in
+	Namespace string
+
+	SpokeK8sClientFactory spoke_k8s_client.SpokeK8sClientFactory
+	SpokeClient           spoke_k8s_client.SpokeK8sClient
 }
 
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=hypershiftagentserviceconfigs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=agent-install.openshift.io,resources=hypershiftagentserviceconfigs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;create;watch
```

Comment:
I don't think we need or want the ability to create CRDs on the host cluster, right?

Reply:
We should figure that out so we can drop this permission on the hub.

Reply:
sure, removed. WIll test again without it.